src/hook.c

```diff
@@ -757,3 +761,46 @@ MH_STATUS WINAPI MH_ApplyQueued(VOID)
         LeaveSpinLock();
     }
 }
+
+//-------------------------------------------------------------------------
+MH_STATUS WINAPI MH_CreateHookApi(LPCSTR pszTarget, LPVOID pDetour, LPVOID *ppOriginal)
+{
+    HMODULE hModule;
+    LPVOID  pTarget;
+    SIZE_T  copySize;
+    LPCSTR  psz = pszTarget;
+    LPCSTR  pszProcName = NULL;
+    CHAR    szModuleName[MAX_PATH];
+
+    if (pszTarget == NULL || pDetour == NULL)
+        return MH_ERROR_INVALID_ARGUMENTS;
+
+    // search the last '.' (replacement for strrchr())
+    while (*psz)
+        if (*psz++ == '.')
+            pszProcName = psz;
```

Comment:
Why first? it searches the last '.' char, and therefore "user32.dll.MessageBoxW" will work as expected.
And about exported functions from headers:
you forgot that:
(DWORD_PTR)&MessageBoxW != (DWORD_PTR)GetProcAddress(GetModuleHandleA("user32.dll"), "MessageBoxW")